src/sys/socket/mod.rs

```diff
@@ -414,6 +414,14 @@ impl<'a> Iterator for CmsgIterator<'a> {
                 Some(ControlMessage::ScmTimestamp(
                     &*(cmsg_data.as_ptr() as *const _)))
             },
+            #[cfg(any(target_os = "linux", target_os = "android", target_os = "macos", target_os = "ios"))]
+            (libc::IPPROTO_IPV6, libc::IPV6_PKTINFO) => unsafe {
```

Comment:
FreeBSD also defines `IPV6_PKTINFO`.  I haven't checked on any of the other BSDs.